REFERENCE.md

```diff
@@ -0,0 +1,1230 @@
+An array of values is also not supported. To match against multiple ICMP
+types, please use separate rules for each ICMP type.
+
+##### `state`
+
+Valid values: INVALID, ESTABLISHED, NEW, RELATED, UNTRACKED
+
+Matches a packet based on its state in the firewall stateful inspection
+table. Values can be:
+
+* INVALID
+* ESTABLISHED
+* NEW
+* RELATED
+* UNTRACKED
+
+##### `ctstate`
+
+Valid values: INVALID, ESTABLISHED, NEW, RELATED, UNTRACKED
+
+Matches a packet based on its state in the firewall stateful inspection
+table, using the conntrack module. Values can be:
+
+* INVALID
+* ESTABLISHED
+* NEW
+* RELATED
+* UNTRACKED
+
+##### `connmark`
+
+Match the Netfilter mark value associated with the packet.  Accepts either of:
+mark/mask or mark.  These will be converted to hex if they are not already.
+
+##### `connlimit_above`
+
+Valid values: %r{^\d+$}
+
+Connection limiting value for matched connections above n.
+
+##### `connlimit_mask`
+
+Valid values: %r{^\d+$}
+
+Connection limiting by subnet mask for matched connections.
+IPv4: 0-32
+IPv6: 0-128
+
+##### `hop_limit`
+
+Valid values: %r{^\d+$}
+
+Hop limiting value for matched packets.
+
+##### `limit`
+
+Rate limiting value for matched packets. The format is:
+rate/[/second/|/minute|/hour|/day].
+
+Example values are: '50/sec', '40/min', '30/hour', '10/day'."
+
+##### `burst`
+
+Valid values: %r{^\d+$}
+
+Rate limiting burst value (per second) before limit checks apply.
+
+##### `uid`
+
+UID or Username owner matching rule.  Accepts a string argument
+only, as iptables does not accept multiple uid in a single
+statement.
+
+##### `gid`
+
+GID or Group owner matching rule.  Accepts a string argument
+only, as iptables does not accept multiple gid in a single
+statement.
+
+##### `match_mark`
+
+Match the Netfilter mark value associated with the packet.  Accepts either of:
+mark/mask or mark.  These will be converted to hex if they are not already.
+
+##### `set_mark`
+
+Set the Netfilter mark value associated with the packet.  Accepts either of:
+mark/mask or mark.  These will be converted to hex if they are not already.
+
+##### `clamp_mss_to_pmtu`
+
+Valid values: `true`, `false`
+
+Sets the clamp mss to pmtu flag.
+
+##### `set_dscp`
+
+Set DSCP Markings.
+
+##### `set_dscp_class`
+
+This sets the DSCP field according to a predefined DiffServ class.
+
+##### `set_mss`
+
+Sets the TCP MSS value for packets.
+
+##### `pkttype`
+
+Valid values: unicast, broadcast, multicast
+
+Sets the packet type to match.
+
+##### `isfragment`
+
+Valid values: `true`, `false`
+
+Set to true to match tcp fragments (requires type to be set to tcp)
+
+##### `recent`
+
+Valid values: set, update, rcheck, remove
+
+Enable the recent module. Takes as an argument one of set, update,
+rcheck or remove. For example:
+
+  If anyone's appeared on the 'badguy' blacklist within
+  the last 60 seconds, drop their traffic, and update the timestamp.
+
+  `firewall { '100 Drop badguy traffic':
```

Comment:
Newlines are not rendering properly. Try using ``` above and below examples. This should preserve comments too so use `#` for those.
```
firewall { '100 Drop badguy traffic':
...
}
```

Reply:
worked